### PR TITLE
Add more intro to table subpackage description and streamline presentation

### LINKED ISSUE #31: tables
The table section could be expanded:

@adrn: Need more introduction to the table subpackage!

@perrygreenfield: Some mention that some of these are inspired by Pandas?

### PR DESCRIPTION
**Pull request by hamogu**
I've added a mention of numpy ans pandas as inspiration for the table
subpackage. Table is not now, and more details can be found in paper 1,
so I think a short mention is enough at this point. I've also removed
intermediate headings, since table is already a subsubsection.

closes #31

### CODE REVIEW

File: main.tex [hunk 1]
```diff
@@ -709,13 +709,20 @@ \subsubsection{Tables}
...
-and manipulating heterogeneous data.
-The package was described in detail in \cite{astropy}.
-Thus, in what follows, we only summarize key new features or updates to
-\package{astropy.table} since the previous \astropy paper.
+The \package{astropy.table} subpackage provides functionality for
+representing and manipulating heterogeneous data. In some respects,
+this is similar to \package{numpy} record arrays \citep{numpy} or
+\package{pandas} dataframes \citep{pandas} but with modifications for
+astronomical data. Most notably, tables from \package{astropy.table}
+allow for table or column metadata and can handle
```
Comment (bsipocz) [inline main.tex:717]: Can we use capitized `Table` and `Column` here?

File: main.tex [hunk 2]
```diff
@@ -709,13 +709,20 @@ \subsubsection{Tables}
...
+vectors or arrays as table entries.
+The package was described in
```
Comment (bsipocz) [inline main.tex]: package -> subpackage

File: main.tex (hunk 1 above)
Comment (hamogu) [reply main.tex:717]: We could, but I did not intend them to mean `Table` as in "instance of the table class", I actually mean: Metadata that is associated with a table (which then in astropy is implemented as a dictionary called "meta" that is an attribute of Table objects).

File: main.tex (hunk 2 above)
Comment (hamogu) [reply main.tex]: will do.

File: main.tex (hunk 1 above)
Comment (hamogu) [reply main.tex:717]: Just to be clear, we could talk about `Table` and `Column` as objects in Python code, but I don't see what we gain from them here. (Saying: "they allow for `Table` and `QTable`, as well as `Column` and `MaskedColumn` metadata" ist just not as compact as saying" table and column", where tables and column used independent of the implementation details.

Comment (hamogu) [reply main.tex:717]: So, I'll leave it as it is for now.

Comment (bsipocz) [reply main.tex:717]: oh, this sounds fine, thanks for clearing it up.

**Review by adrn — APPROVED** [review]

**Pull request merged**